src/compute/src/server.rs

```diff
@@ -347,23 +345,31 @@ pub async fn compute_node_serve(
 }
 
 /// Check whether the compute node has enough memory to perform computing tasks. Apart from storage,
-/// it must reserve at least `MIN_COMPUTE_MEMORY_MB` for computing and `SYSTEM_RESERVED_MEMORY_MB`
-/// for other system usage. Otherwise, it is not allowed to start.
-fn validate_compute_node_memory_config(cn_total_memory_bytes: usize, storage_memory_bytes: usize) {
+/// it is recommended to reserve at least `MIN_COMPUTE_MEMORY_MB` for computing and
+/// `SYSTEM_RESERVED_MEMORY_MB` for other system usage. If the requirement is not met, we will print
+/// out a warning and enforce the memory used for computing tasks as `MIN_COMPUTE_MEMORY_MB`.
+fn validate_compute_node_memory_config(
+    cn_total_memory_bytes: usize,
+    storage_memory_bytes: usize,
+) -> usize {
     if storage_memory_bytes > cn_total_memory_bytes {
-        panic!(
-            "The storage memory exceeds the total compute node memory:\nTotal compute node memory: {}\nStorage memory: {}\nAt least 1 GiB memory should be reserved apart from the storage memory. Please increase the total compute node memory or decrease the storage memory in configurations and restart the compute node.",
+        tracing::warn!(
+            "The storage memory exceeds the total compute node memory:\nTotal compute node memory: {}\nStorage memory: {}\nWe recommend that at least 1 GiB memory should be reserved apart from the storage memory. Please increase the total compute node memory or decrease the storage memory in configurations.",
```

Comment:
Trying to be more friendly for end users.

```suggestion
            "The storage memory exceeds the total compute node memory:\nTotal compute node memory: {}\nStorage memory: {}\nWe recommend that at least 4 GiB memory should be reserved for RisingWave. Please increase the total compute node memory or decrease the storage memory in configurations.",
```